fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFab.kt

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.FloatingActionButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Floating action button for tabs tray.
+ *
+ * @param tabsTrayStore [TabsTrayStore] used to listen for changes to [TabsTrayState].
+ */
+@Composable
+fun TabsTrayFab(
+    tabsTrayStore: TabsTrayStore,
+) {
+    val trayState = tabsTrayStore
+        .observeAsComposableState { state ->
+            TrayState(
+                state.selectedPage,
+                state.syncing,
+                state.mode == TabsTrayState.Mode.Normal,
+            )
+        }.value ?: TrayState()
```

Comment:
do we benefit from having a wrapper for the selected page, syncing, and browsing mode vs observing these variables individually?

Reply:
We don't really, I removed it and added individual variables.